pkg/dbmate/db.go

```diff
@@ -202,6 +218,23 @@ func doTransaction(db *sql.DB, txFunc func(Transaction) error) error {
 	return tx.Commit()
 }
 
+func transactAndRollback(db *sql.DB, txFunc func(Transaction) error) error {
+	tx, err := db.Begin()
```

Comment:
db could be nil, this would panic

Reply:
Good point, but this function is used only from one location where param is guaranteed to be not nil. I am not familiar with golang patterns here, but if we start checking all param in each level, wouldn't that lead to lot of duplicate checks? :thinking:

Reply:
:man_shrugging: